scripts/migrate-attachments.pl

```diff
@@ -185,25 +270,32 @@
 }
 
 elsif ($options{delete}) {
-  my $storage = $options{delete};
-  my ($total)
-    = $dbh->selectrow_array(
-    "SELECT COUNT(*) FROM attachments WHERE attach_size != 0");
+  my $source = $options{delete};
+
+  my ($total) = $dbh->selectrow_array('
+    SELECT COUNT(*) 
+      FROM attachments 
+           JOIN attachment_storage_class ON attachments.attach_id = attachment_storage_class.id 
+     WHERE attachment_storage_class.storage_class = ? AND attachments.attach_size != 0',
+    undef, $source);
+
   confirm(sprintf 'DELETE %s attachments from %s?', $total, $options{delete});
 
-  my $sth
-    = $dbh->prepare(
-    "SELECT attach_id FROM attachments WHERE attach_size != 0 ORDER BY attach_id DESC"
-    );
-  $sth->execute();
+  my $sth = $dbh->prepare('
```

Comment:
`$sth` is an ambiguous variable name.

Reply:
$sth means "statement handle" and $dbh means "database handle". The statement handle is so we can iterate over the rows as they are returned from the database when executing a query. Instead of pulling in all rows at once into memory. $sth is used throughout the BMO code and this script will not be used very often.